### PR TITLE
Add Asteroid Explosion OnCollision & Destroy GettingSucked Audios & 3D Sound

### PR DESCRIPTION
**Pull request by PyroWilDx**
- Changed Spawner to singleton class.
- Created BlackHole singleton class.
- Added explosion when an Asteroid is collided + explosion sound.
- Destroy GettingSucked audio instances after they finished playing.
- Add 3D sounds.

### CODE REVIEW

File: Assets/Prefabs/Object/Astronaut.prefab [hunk 1]
```diff
@@ -97,8 +97,8 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: 3e8cea98d335472ea8eaf2963c03c09e, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
-  minMoveSpeed: 0.2
-  maxMoveSpeed: 0.4
```
Comment (shogo0x2e) [inline Assets/Prefabs/Object/Astronaut.prefab:101]: ask 
Is this change intended? I think we decided to go with this number 0.2 - 0.4.
(But of course we can make changes after this PR merged)

File: Assets/Prefabs/Object/Asteroid.prefab [hunk 2]
```diff
@@ -94,10 +94,14 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: 9be033a3997445ce822eb199ad904517, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
-  minMoveSpeed: 0.1
-  maxMoveSpeed: 0.12
```
Comment (shogo0x2e) [inline Assets/Prefabs/Object/Asteroid.prefab:98]: ask 
Is this change intended? I think we decided to go with this number 0.1 - 0.12.
(But of course we can make changes after this PR merged)

File: Assets/Scripts/BlackHole.cs
Comment (shogo0x2e) [inline Assets/Scripts/BlackHole.cs:1]: good 
Nice to use singleton!

File: Assets/Scripts/Object/AbstractObject.cs [hunk 3]
```diff
@@ -1,36 +1,36 @@
...
         private Vector3 rotDirection;
-        // Reference to Spawner.cs
-        private Spawner spawner;
 
+        private bool sucked = false;
+        
+        [SerializeField] private AudioSource onDestroySound;
+
+        public void Start() {
+            mainMode = BlackHole.GetInstance() != null;
```
Comment (shogo0x2e) [inline Assets/Scripts/Object/AbstractObject.cs:27]: ask 
What `mainMode` used for? Do you envision implementing a start screen or something in the future development?

File: Assets/Scripts/Object/Spawnable/Asteroid.cs [hunk 4]
```diff
@@ -15,7 +21,20 @@ public override void OnSpawn() {
             currentModel.transform.parent = transform;
         }
 
-        public void Explode(Vector3 colPosition, float colForce) {
+        public void OnCollisionEnter(Collision collision) {
+            if (collision.gameObject.GetComponent<AbstractObject>() != null) {
+                return; // Do not explode when colliding with other space objects
```
Comment (shogo0x2e) [inline Assets/Scripts/Object/Spawnable/Asteroid.cs:26]: good 
Nice documentation

**Review by shogo0x2e — COMMENTED** [review]
I left some question! And I want you to confirm when the asteroid explosion animation and sound effect will play? I pulled your changes but I can't try those 🥺

File: Assets/Scripts/Object/AbstractObject.cs (hunk 3 above)
Comment (PyroWilDx) [reply Assets/Scripts/Object/AbstractObject.cs:27]: for now it's just useful for test scenes (because sometime there's no blackhole in test scenes), but yeah maybe it could also be useful in the future

File: Assets/Prefabs/Object/Astronaut.prefab (hunk 1 above)
Comment (PyroWilDx) [reply Assets/Prefabs/Object/Astronaut.prefab:101]: in PR-#26 https://github.com/shogo0x2e/BlackHoleBoogie/pull/26/files, I removed the line ```moveSpeed = moveSpeed + 0.01F; // A bit of acceleration``` (I don't know who added that), and I believe it's this line that was making objects very fast. So I increased their speed a little, but yeah we can change after this PR if it's too fast

File: Assets/Prefabs/Object/Asteroid.prefab (hunk 2 above)
Comment (PyroWilDx) [reply Assets/Prefabs/Object/Asteroid.prefab:98]: same as above

**Review by shogo0x2e — APPROVED** [review]
Looks good to me!

**Comment by PyroWilDx** [comment]
> Looks good to me!

thanks for checking this PR :)
